client/state/sharing/keyring/reducer.js

```diff
@@ -28,11 +28,11 @@ export const isFetching = createReducer( false, {
 // Stores the list of available keyring connections
 export const items = createReducer( {}, {
 	[ KEYRING_CONNECTIONS_RECEIVE ]: ( state, { connections } ) => ( { ...keyBy( connections, 'ID' ) } ),
-	[ PUBLICIZE_CONNECTION_CREATE ]: ( state, { ID, siteId } ) => ( { ...state, [ ID ]: {
-		...state[ ID ], sites: [ ...state[ ID ].sites, siteId.toString() ]
+	[ PUBLICIZE_CONNECTION_CREATE ]: ( state, { connection: { keyring_connection_ID: ID, site_ID } } ) => ( { ...state, [ ID ]: {
+		...state[ ID ], sites: [ ...state[ ID ].sites, site_ID.toString() ]
```

Comment:
> Is site_ID not always a fixed type?
The `/me/keyring-connections` endpoint returns the site IDs as an array of strings. While `user_ID` gets treated with an `absint()` before being returned, site IDs do not.
Also see https://github.com/Automattic/wp-calypso/pull/9068#discussion_r86141237